src/hotspot/share/opto/parse1.cpp

```diff
@@ -507,34 +493,44 @@ Parse::Parse(JVMState* caller, ciMethod* parse_method, float expected_uses)
 
   // Do some special top-level things.
   if (depth() == 1 && C->is_osr_compilation()) {
+    _tf = C->tf();     // the OSR entry type is different
     _entry_bci = C->entry_bci();
     _flow = method()->get_osr_flow_analysis(osr_bci());
-    if (_flow->failing()) {
-      assert(false, "type flow analysis failed for OSR compilation");
-      C->record_method_not_compilable(_flow->failure_reason());
+  } else {
+    _tf = TypeFunc::make(method());
+    _entry_bci = InvocationEntryBci;
+    _flow = method()->get_flow_analysis();
+  }
+
+  if (_flow->failing()) {
+    assert(false, "type flow analysis failed during parsing");
+    C->record_method_not_compilable(_flow->failure_reason());
 #ifndef PRODUCT
       if (PrintOpto && (Verbose || WizardMode)) {
-        tty->print_cr("OSR @%d type flow bailout: %s", _entry_bci, _flow->failure_reason());
+        if (is_osr_parse()) {
+          tty->print("OSR @%d ", _entry_bci);
+        }
+        tty->print_cr("type flow bailout: %s", _flow->failure_reason());
```

Comment:
Not sure if we want to keep the single `print_cr` for log atomicity reasons. I think this would be good too:


```
if (is_osr_parse()) {
  tty->print_cr("OSR @%d type flow bailout: %s", _entry_bci, _flow->failure_reason());
} else {
  tty->print_cr("type flow bailout: %s", _flow->failure_reason());
}
```